enhancements/single-node-production-edge-cluster-profile.md

```diff
@@ -0,0 +1,380 @@
+`MachineConfig` choice as per
+[enhancements/support-for-realtime-kernel](https://github.com/openshift/enhancements/blob/master/enhancements/support-for-realtime-kernel.md). *See open questions*
+
+### User Stories
+
+#### As a user, I can deploy OpenShift in a supported single-node configuration
+
+A user will be able to run the OpenShift installer to create a single-node
+deployment, with some limitations (see non-goals above). The user
+will not require special support exceptions to receive technical assistance
+for the features supported by the configuration.
+
+### Implementation Details/Notes/Constraints [optional]
+
+Some OpenShift components (such as Etcd and Ingress) require
+a minimum of 2 or 3 nodes. The `single-node-production-edge`
+cluster profile will configure these components as appropriate
+for a single node.
+
+Known gaps:
+- `cluster-etcd-operator` will not deploy the etcd cluster without minimum of 3 master nodes (can be changed by enabling `useUnsupportedUnsafeNonHANonProductionUnstableEtcd`)
+- Even with the unsupported feature flag, `etcd-quorum-guard` still requires 3 nodes due to its replica count.
+- `cluster-authentication-operator` will not deploy `OAuthServer` without minimum of 3 master nodes (can be change by enabling `useUnsupportedUnsafeNonHANonProductionUnstableOAuthServer`)
+- `cluster-ingress-operator` deploys the router with 2 replicas. On a single node one will fail to start and the ingress will show as degraded.
+
+When we are deploying a cluster with the `single-node-production-edge`
+cluster profile the relevant operators should support a non-HA
+configuration that makes the correct adjustments to the deployment
+(e.g., `cluster-ingress-operator` should deploy a single router,
+`cluster-etcd-operator` should deploy the `etcd-member`
+[without waiting for 3 master nodes](https://github.com/openshift/cluster-etcd-operator/blob/98590e6ecfe282735c4eff01432ae40b29f81202/pkg/etcdenvvar/etcd_env.go#L72))
+
+In addition, some components are not relevant for this cluster
+profile (e.g. console, cluster-autoscaler, marketplace?) and shouldn't
+be deployed at all.
+
+
+### Risks and Mitigations
+
+*What are the risks of this proposal and how do we mitigate. Think broadly. For
+example, consider both security and how this will impact the larger OKD
+ecosystem.*
+
+*How will security be reviewed and by whom? How will UX be reviewed and by whom?*
+
+## Design Details
+
+### Open Questions
+
+1. Telco workloads frequently require a realtime kernel. How will a
+   user specify whether to use the realtime or regular kernel? Should
+   we assume they always want the realtime version?
+2. Similarly, telco workloads typically require special network setups
+   for a host to boot, including bonded interfaces, access to multiple
+   VLANs, and static IPs. How do we anticipate configuring those?
+3. The machine-config-operator works by (almost always) rebooting a host.
```

Comment:
No. MCO is absolutely necessary as it is used by PAO (optional OLM operator) and NTO to compute the proper host OS and kernel tuning values. It is also used to allocate hugepages for example.

Reply:
Btw, you either need to describe a way to pre-deploy an image including optional OLM operators and reboots or allow day-2 operations that do the same. Not all Telco deployments are the same (RT vs. non-RT, different networking, different CNFs, different hugepages, different NUMA topology...).